corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js

```diff
@@ -274,6 +274,35 @@ describe('Entries', function () {
         assert.isTrue(spy.calledOnce);
     });
 
+    it('Should return EthiopanDateEntry', function () {
+        questionJSON.datatype = Const.DATE;
+        questionJSON.answer = '2021-01-29'; // 2013-05-21 in Ethiopian
+        questionJSON.style = { raw: 'ethiopian' };
+
+        var entry = UI.Question(questionJSON).entry;
+        entry.entryId = 'fake-div';
+        assert.isTrue(entry instanceof Controls.EthiopianDateEntry);
+        assert.equal(entry.templateType, 'ethiopian-date');
+
+        $(document.body).append("<div id='"+ entry.entryId + "'></div>");
```

Comment:
I didn't see any other tests adding items to the DOM elsewhere, but I wanted to be able to test interaction with the widget. Would love feedback on whether this is a bad idea.

Reply:
It's more typical to put DOM for tests in the `fixtures` block in the mocha template, would that work here? I'm curious, what is it that doesn't work if the question isn't actually part of the DOM? Do the `entry.$picker` functions not work?

Reply:
Oh that's what I was looking for, thanks! Yes, it's the `entry.$picker.calendarsPicker` functions that don't work. I think this is in large part due to how this calendar widget is written as a jquery plugin.

Reply:
cool, I can imagine the plugin designers assuming they're on a real page